### ISSUE TITLE
arviz.summary() indexes parameters from 0, instead of taking the index described in coordinates

### ISSUE THREAD

**Issue by orbidder** [arviz-devs/arviz#1499]
**Describe the bug**
Stan indexes from 1. When calling arviz.summary(), the parameters are indexed from 0, regardless of the coordinates specified in the arviz.from_pystan() call.

**To Reproduce**
```
X = np.random.rand(100,9)
y = (X*0.1).sum(axis=1) + np.random.normal(0,1,100)

reg_data = {'X': X,
            'y': y,
            'N': X.shape[0],
            'K': X.shape[1]}
reg_code = """
data {
    int<lower=0> N; // number of samples
    int<lower=0> K; // number of predictors
    vector[N] y; // response variable
    matrix[N, K] X; // predictor matrix
}
parameters {
    real alpha;
    vector[K] beta; // coefficients for predictors
    real<lower=0> sigma;
}
transformed parameters {
    vector[N] mu;
    mu = alpha + X * beta;
}
model {
    y ~ normal(mu, sigma);
}
generated quantities {
    vector[N] log_lik;
    vector[N] y_hat;
    
    for (i in 1:N) {
        log_lik[i] = normal_lpdf(y[i] | mu[i], sigma);
        y_hat[i] = normal_rng(mu[i], sigma);
    }

}
"""
# fit the model
sm = ps.StanModel(model_code=reg_code)
fit = sm.sampling(data=reg_data, iter=5000, chains=4)

# convert pystan to arviz
fit_az = az.from_pystan(posterior=fit,
                      posterior_predictive="y_hat",
                      observed_data=["y"],
                      log_likelihood={"y": "log_lik"},
                      coords={"chain": np.arange(4),
                              "draw": np.arange(2500),
                              "var": np.arange(1,X.shape[1]+1)},
                      dims={'alpha': ["chain","draw"],
                            'beta': ["chain","draw", "var"],
                            'sigma': ["chain","draw"]})

# plot_forest() correctly inherits the labels from the fit_az object
axes = az.plot_forest(
    fit_az,
    kind="ridgeplot",
    var_names=["alpha", "beta", "sigma"],
    combined=True,
    figsize=(9, 4),
    hdi_prob=0.95
)
plt.show()

# but summary just indexes the betas from 0, even though stan calls them beta[1], beta[2]...beta[9]
az.summary(fit_az, var_names=["alpha", "beta", "sigma"])
```

**Expected behavior**
az.summary() should number the parameters 1-9, as is specified in the coords, in the same way that plot_forest does.

**Additional context**
I think this is particular to pystan, because other packages may just index from 0 like python does.

**Comment by canyon289** [comment]
This sounds very reasonable and I appreciate you pointing it out!

**Comment by OriolAbril** [comment]
I think #1201 would fix this. Maybe we should rebase and try to get it through the finish line.

**Comment by orbidder** [comment]
> This sounds very reasonable and I appreciate you pointing it out!

No problem!

**Comment by utkarsh-maheshwari** [comment]
Hey, I wanted to contribute. 
Are the required changes done in #1201 to solve this issue?

**Comment by utkarsh-maheshwari** [comment]
Hey, I want to contibute. 
Are the required changes done in #1201 to solve this issue?

**Comment by OriolAbril** [comment]
Let me see if I have time to finish #1201 in the next 1-2 weeks before working on that. As #1201 changes most of how labels work, if this issue is fixed only to have #1201 merged some weeks later the work done to fix this issue only would not be used at all

**Comment by OriolAbril** [comment]
Just merged the labeling restructure PR which fixes that and greatly extends the labeling options in ArviZ plots, docs are still wip but https://arviz-devs.github.io/arviz/user_guide/label_guide.html should be a good overview, let me know what you think.

Note that changes are only in development version for now, they have not been released yet
